clang/test/Driver/sycl-intelfpga-aoco.cpp

```diff
@@ -0,0 +1,95 @@
+// UNSUPPORTED: system-windows
+
+/// -fintelfpga static lib (aoco)
+// RUN:  echo "Dummy AOCO image" > %t.aoco
+// RUN:  echo "void foo() {}" > %t.c
+// RUN:  echo "void foo2() {}" > %t2.c
+// RUN:  %clang -c -o %t.o %t.c
+// RUN:  %clang -fsycl -c -o %t2.o %t2.c
+// RUN:  %clang_cl -fsycl -c -o %t2_cl.o %t2.c
+// RUN:  clang-offload-wrapper -o %t-aoco.bc -host=x86_64-unknown-linux-gnu -kind=sycl -target=fpga_aoco-intel-unknown-sycldevice %t.aoco
+// RUN:  llc -filetype=obj -o %t-aoco.o %t-aoco.bc
+// RUN:  clang-offload-wrapper -o %t-aoco_cl.bc -host=x86_64-unknown-linux-gnu -kind=sycl -target=fpga_aoco-intel-unknown-sycldevice %t.aoco
+// RUN:  llc -filetype=obj -o %t-aoco_cl.o %t-aoco_cl.bc
+// RUN:  llvm-ar crv %t_aoco.a %t.o %t2.o %t-aoco.o
+// RUN:  llvm-ar crv %t_aoco_cl.a %t.o %t2_cl.o %t-aoco_cl.o
+// RUN:  %clangxx -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-device-lib=all -fintelfpga %t_aoco.a %s -### -ccc-print-phases 2>&1 \
+// RUN:  | FileCheck -check-prefix=CHK-FPGA-AOCO-PHASES %s
+// CHK-FPGA-AOCO-PHASES: 0: input, "[[INPUTA:.+\.a]]", object, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 1: input, "[[INPUTCPP:.+\.cpp]]", c++, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 3: input, "[[INPUTCPP]]", c++, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 4: preprocessor, {3}, c++-cpp-output, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 5: compiler, {4}, sycl-header, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 6: offload, "host-sycl (x86_64-unknown-linux-gnu)" {2}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {5}, c++-cpp-output
+// CHK-FPGA-AOCO-PHASES: 7: compiler, {6}, ir, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 8: backend, {7}, assembler, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 9: assembler, {8}, object, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 10: linker, {0, 9}, image, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 11: compiler, {4}, ir, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 12: input, "[[INPUTA]]", archive
+// CHK-FPGA-AOCO-PHASES: 13: partial-link, {9, 12}, object
+// CHK-FPGA-AOCO-PHASES: 14: clang-offload-unbundler, {13}, object
+// CHK-FPGA-AOCO-PHASES: 15: linker, {11, 14}, ir, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 16: sycl-post-link, {15}, ir, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 17: llvm-spirv, {16}, spirv, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 18: input, "[[INPUTA]]", archive
+// CHK-FPGA-AOCO-PHASES: 19: clang-offload-unbundler, {18}, fpga_dependencies_list
+// CHK-FPGA-AOCO-PHASES: 20: input, "[[INPUTA]]", fpga_aoco
+// CHK-FPGA-AOCO-PHASES: 21: clang-offload-unbundler, {20}, fpga_aoco
+// CHK-FPGA-AOCO-PHASES: 22: backend-compiler, {17, 19, 21}, fpga_aocx, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 23: clang-offload-wrapper, {22}, object, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 24: offload, "host-sycl (x86_64-unknown-linux-gnu)" {10}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {23}, image
+
+/// FPGA AOCO Windows phases check
```

Comment:
Would these really be needed in a Linux-specific file?

Reply:
It tests the use of `clang-cl` which is allowed on the Linux side of testing.  So while it's not really necessary to test on Linux, it does provide some testing for the Windows path for those who do not do testing on Windows when doing local validation of changes.